Dogfood hiding secrets added via YAML

noahtalerman opened the pull request:
Dogfooding for the improvement shipped in this user story:
- #23238

noahtalerman commented:
> TODO: Before merge, update the name of the Chrome secret in the fleetdm/fleet repo's variables to `FLEET_SECRET_MANAGED_CHROME_ENROLLMENT_TOKEN`

@allenhouchins can you please help me with this update to the fleetdm/fleet repo variables? (I think you have access)

When you make the change, please mark the PR as ready for review, approve, and merge.

allenhouchins commented:
Hey @noahtalerman just wanting to make sure we are on the same page. I do not have access to update GitHub secrets (that would need to be handled by @lukeheath) but we shouldn't need to update the name of the secret in GitHub in order to implement this change since we are using GitOps to manage dogfood and secrets. In addition to the change you made in the profile, your PR should include updating [this line](https://github.com/fleetdm/fleet/blob/e80034e5e478757badd1802c5bc417ef2b52ded1/.github/workflows/dogfood-gitops.yml#L78) to `FLEET_SECRET_MANAGED_CHROME_ENROLLMENT_TOKEN: ${{ secrets.CLOUD_MANAGEMENT_ENROLLMENT_TOKEN }}`

This might not be the best example since secret substitution has always been built into GitOps. This might be a better thing to demonstrate exclusively through the UI/API -- add a secret via the API, upload a script via the UI, scope it to a device, and show the results.

I also wanted to double check and make sure we have tested the user experience of removing and re-adding that profile before we merge this update. I mainly want to make sure the change is silent and the user isn't logged out of Chrome or receiving any other prompts. 

As a complete aside, and admittedly I might not be fully understanding this feature but now that I've dug into "secret variables", calling them secret variables is a little misleading. What we have implemented is more akin to secret/string substitution. Variables indicates that I can write it once and the value will change in realtime. For example, $HOSTNAME being different on every device or $CURRENT_USER being the result of a command running on a device to determine the currently logged in user. I understand the value of the "secret variable" can change by updating it through the API but that's still just a manual string substitution. Just feedback for thought. No actions required.

lukeheath commented:
I imagine @noahtalerman is changing the secret to match other dogfood-related secrets. I can make the change once this merges. We can discuss at dogfooding sync.

noahtalerman commented:
@lukeheath @allenhouchins this PR is to dogfood this improvement: hide secrets added via YAML (GitOps) in scripts or configuration profiles (#23238)

Looks like we don't actually need to update the secret name in the repo. Instead, I updated it's reference [here](https://github.com/fleetdm/fleet/pull/25384/files#diff-2788b5ca8ed07f97d0c1b76a9574e49ff54a35d3dab8552ee0b31f362a420065R78).

We added the ability for secrets to show up "masked" (not plain text) when you add them via GitOps and later download the profiles/scripts in the UI.

allenhouchins commented:
> Looks like we don't actually need to update the secret name in the repo. Instead, I updated it's reference [here](https://github.com/fleetdm/fleet/pull/25384/files#diff-2788b5ca8ed07f97d0c1b76a9574e49ff54a35d3dab8552ee0b31f362a420065R78).

@noahtalerman Perfect! I created a separate issue to track getting all the other secrets up to date so they follow the `FLEET_SECRET` format to make it clearer as people browse our repo. fleetdm/confidential#9349

noahtalerman commented:
> make sure we have tested the user experience of removing and re-adding that profile before we merge this update. I mainly want to make sure the change is silent and the user isn't logged out of Chrome or receiving any other prompts.

@allenhouchins you raise a great point here. I think we yolo merge this PR and monitor the end user experience. Quickest way to test.

allenhouchins reviewed (approved):
@noahtalerman Looks good. We have a heavy demo day today so we will need to wait to do the Merge until after hours. I'm happy to do that if you'd like. Just let me know.

noahtalerman commented:
> We have a heavy demo day today so we will need to wait to do the Merge until after hours. I'm happy to do that if you'd like. Just let me know.

@allenhouchins roger that. Planning on merging at 5:30p EST.

noahtalerman commented:
> Planning on merging at 5:30p EST.

Whoops need a codeowner approval. 

@lukeheath, when you get the chance, can you please review this PR? Please don't merge it. I want to merge so that I can monitor the profile redelivery / end user experience in dogfood.

Thanks!
